bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestBot.java

```diff
@@ -72,6 +73,8 @@ class PullRequestBot implements Bot {
     private final boolean jcheckMerge;
     private final Set<String> mergeSources;
     private final boolean enableBackport;
+    private final Map<String, List<String>> issuePRMap;
+    private final Map<String, Boolean> initializedPRs = new ConcurrentHashMap<>();
```

Comment:
This could just be a `Set`. Also it doesn't need to be concurrent as it's local to just this PullRequestBot, and we don't run WorkItems for the same PR in parallel.

Reply:
Ah, right! Thanks!

Reply:
Think again about this, It's possible that CheckWorkItems for different PRs in the same repo run in parallel. Every CheckWorkItem is possible to modify `initializedPRs`. So we still need synchronize it.

Reply:
You are right, the map will be concurrently modified, just not the same element, and that requires a `ConcurrentHashMap` or synchronization to avoid corrupting data. I think `ConcurrentHashMap` is preferred here.

Reply:
Yes, I will change it back